oresat_configs/scripts/gen_dcf.py

```diff
@@ -1,15 +1,46 @@
 """Generate a DCF for from an OreSat card's object directory."""
 
-import sys
-from argparse import ArgumentParser
+from argparse import ArgumentParser, Namespace
 from datetime import datetime
+from typing import Optional
 
 import canopen
 
-from .. import OreSatConfig, OreSatId
+from .. import Consts, OreSatConfig
 
 GEN_DCF = "generate DCF file for OreSat node(s)"
-GEN_DCF_PROG = "oresat-gen-dcf"
+
+
+def build_parser(parser: ArgumentParser) -> ArgumentParser:
```

Comment:
Is there a reason this exists as duplicate code across all of the OD sub-parsers?

Could this be moved to the root level of the cmd line tool?

I don't forsee a use case where we built an xtce generator but not the DCF generator.

Reply:
`pyproject.toml` installs each subscript as both a standalone script and as a subscript to the root level `oresat-configs` (the `[project.scripts]` section). Because they need to be able to run without the root level tool, they need to construct their own arguments on their own. We could have another file `script-args.py` that held the shared code and have both the root level and subscripts import from that, but a bit of copy/pasting seemed simple enough.

For example, I very much wanted to put the `--oresat` argument in the root level because every script takes it. If I were to then run `oresat-gen-dcf` though, the standalone version, it wouldn't have the `--oresat` option because it bypassed the root level.

If we're OK with not having the standalone versions of the scripts, that would allow the implementation to de-duplicate a bunch of the arguments up into the root level. I didn't get rid of them here to keep backwards compatibility.